Main.py

```diff
@@ -142,7 +143,69 @@ def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = No
+                    for starting_item, count in world.start_inventory[player].value.items():
+                        custom_item_pool[replacement] = custom_item_pool.setdefault(replacement, 0) + count
+                        custom_item_pool[starting_item] = 0
+                elif replacement == "start_inventory":
+                    for _ in range(custom_item_pool.setdefault(item_name, 0)):
+                        world.push_precollected(world.create_item(item_name, player))
+                    custom_item_pool[item_name] = 0
+                else:
+                    custom_item_pool[replacement] = custom_item_pool.setdefault(replacement, 0) + custom_item_pool.setdefault(item_name, 0)
+                    custom_item_pool[item_name] = 0
+            for item_name, custom_pool_count in custom_item_pool.items():
+                player_pool_count = player_item_pool.setdefault(item_name, 0)
+                if custom_pool_count > player_pool_count:
+                    for _ in range(0, custom_pool_count - player_pool_count):
+                        if item_name is None:
+                            item = world.worlds[player].create_filler()
+                        else:
+                            item = world.create_item(item_name, player)
+                        world.itempool.append(item)
+                elif custom_pool_count < player_pool_count:
+                    item = world.create_item(item_name, player)
+                    for _ in range(0, player_pool_count - custom_pool_count):
+                        try:
+                            world.itempool.remove(item)
+                        except ValueError:
+                            logging.warning(f"Could not remove {item.name} for {world.player_name[player]} for custom item pool, item does not exist")
+                            break
+            if world.custom_item_pool[player].value.setdefault('item_pool_correction', True):
+                custom_pool_size = sum(custom_item_pool.values())
+                player_pool_size = sum(player_item_pool.values())
+                world.random.shuffle(world.itempool)
+                diff = custom_pool_size - player_pool_size
+                if diff > 0:
+                    removed_items = 0
+                    for item in world.itempool:
+                        if item.player == player and item.classification == ItemClassification.filler:
+                            world.itempool.remove(item)
+                            removed_items += 1
+                            if removed_items == diff:
+                                break
+                    else:
+                        logging.warning(f"Unable to remove enough filler items for player {world.player_name[player]} to correct item pool size")
+                        logging.warning(f"Player {world.player_name[player]}'s item pool oversized by {diff - removed_items} items")
+                for _ in range(diff, 0):
```

Comment:
If you have `use_defaults` disabled, this loop goes through and creates an extra item without removing an item so you end up with more items than locations before the generator goes to fill the locations. This may not affect much in solo seeds, as the generator will guarantee that progression items are still in the pool, but if you play with others, the generator may not be able to place others' items which is a big **no** in my book. Any wacky settings you do to your item pool should not affect which items others can access. See below for example output between a Rogue Legacy seed (with custom item pool) and a Link to the Past seed.

```yaml
Rogue Legacy:
  custom_item_pool: # Customize your item pool. Use with care
    set:
      Death Defiance: 5
    use_defaults: false
```

```
nplaced items(143): [5000 Gold (Phar1), Down Strike Up (Phar1), Siphon Runes (Phar1), 3000 Gold (Phar1), Rupees (20) (ALTTP1), 5000 Gold (Phar1), Rupees (20) (ALTTP1), Invulnerability Time Up (Phar1), Triple Stat Increase (Phar1), 5000 Gold (Phar1), 1000 Gold (Phar1), 5000 Gold (Phar1), 3000 Gold (Phar1), Guardian Blueprints (Phar1), Rupees (50) (ALTTP1), Ranger Blueprints (Phar1), 5000 Gold (Phar1), Balance Runes (Phar1), 1000 Gold (Phar1), Bombs (3) (ALTTP1), Triple Stat Increase (Phar1), 3000 Gold (Phar1), Rupees (20) (ALTTP1), 3000 Gold (Phar1), Vault Runes (Phar1), 3000 Gold (Phar1), Traitors (Phar1), 3000 Gold (Phar1), 3000 Gold (Phar1), Triple Stat Increase (Phar1), Bombs (3) (ALTTP1), Single Arrow (ALTTP1), Rupees (20) (ALTTP1), Crit Chance Up (Phar1), 5000 Gold (Phar1), Rupees (20) (ALTTP1), Progressive Spellthieves (Phar1), Rupees (5) (ALTTP1), Sage Blueprints (Phar1), 3000 Gold (Phar1), Rupees (20) (ALTTP1), 5000 Gold (Phar1), 5000 Gold (Phar1), 5000 Gold (Phar1), Crit Damage Up (Phar1), Progressive Spellthieves (Phar1), Crit Chance Up (Phar1), Rupees (20) (ALTTP1), 1000 Gold (Phar1), Arrows (10) (ALTTP1), 5000 Gold (Phar1), Bombs (3) (ALTTP1), Bombs (3) (ALTTP1), Progressive Knights (Phar1), 1000 Gold (Phar1), Triple Stat Increase (Phar1), Rupees (300) (ALTTP1), 1000 Gold (Phar1), 3000 Gold (Phar1), 1000 Gold (Phar1), Dragons (Phar1), Triple Stat Increase (Phar1), Bombs (3) (ALTTP1), 1000 Gold (Phar1), Triple Stat Increase (Phar1), Randomize Children (Phar1), 5000 Gold (Phar1), 5000 Gold (Phar1), Rupees (20) (ALTTP1), Triple Stat Increase (Phar1), Rupees (20) (ALTTP1), 1000 Gold (Phar1), 1000 Gold (Phar1), Arrows (10) (ALTTP1), 1000 Gold (Phar1), Haste Runes (Phar1), 3000 Gold (Phar1), Triple Stat Increase (Phar1), Curse Runes (Phar1), Triple Stat Increase (Phar1), 1000 Gold (Phar1), Rupees (20) (ALTTP1), 3000 Gold (Phar1), Progressive Mages (Phar1), 3000 Gold (Phar1), 5000 Gold (Phar1), Rupees (20) (ALTTP1), 3000 Gold (Phar1), Bombs (3) (ALTTP1), 5000 Gold (Phar1), 3000 Gold (Phar1), Arrows (10) (ALTTP1), 5000 Gold (Phar1), Rupee (1) (ALTTP1), Rupees (50) (ALTTP1), 5000 Gold (Phar1), Rupees (20) (ALTTP1), 3000 Gold (Phar1), Rupees (5) (ALTTP1), 1000 Gold (Phar1), 3000 Gold (Phar1), Rupees (5) (ALTTP1), 1000 Gold (Phar1), Blood Blueprints (Phar1), Progressive Miners (Phar1), 5000 Gold (Phar1), 3000 Gold (Phar1), Triple Stat Increase (Phar1), Death Defiance (Phar1), Arrows (10) (ALTTP1), Crit Damage Up (Phar1), Progressive Mages (Phar1), Bombs (3) (ALTTP1), Crit Damage Up (Phar1), Rupees (20) (ALTTP1), Rupees (20) (ALTTP1), Death Defiance (Phar1), 1000 Gold (Phar1), 5000 Gold (Phar1), Rupees (300) (ALTTP1), 1000 Gold (Phar1), Rupees (20) (ALTTP1), Rupees (20) (ALTTP1), 1000 Gold (Phar1), Triple Stat Increase (Phar1), Triple Stat Increase (Phar1), 1000 Gold (Phar1), 1000 Gold (Phar1), 3000 Gold (Phar1), Rupees (20) (ALTTP1), 5000 Gold (Phar1), 5000 Gold (Phar1), 1000 Gold (Phar1), 5000 Gold (Phar1), 1000 Gold (Phar1), Bombs (3) (ALTTP1), 1000 Gold (Phar1), Crit Chance Up (Phar1), Vampire Runes (Phar1), 1000 Gold (Phar1), 1000 Gold (Phar1), Progressive Knaves (Phar1), Triple Stat Increase (Phar1)] - Unfilled Locations(0): []
```

This whole function should keep the same item pool size or it will affect other seeds placement logic.